### PR TITLE
Add MultiversX namespace

### PR DESCRIPTION
**Pull request by radumojic**
Added specs for CAIP-2 and CAIP-10 on the MultiversX blockchain

### CODE REVIEW

File: mvx/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+namespace-identifier: mvx
+title: MultiversX Namespace
+author: MultiversX Team <contact@multiversx.com>
+discussions-to: https://github.com/ChainAgnostic/namespaces/issues/89
+status: Draft
+type: Informational
+created: 2023-09-11
+requires: ["CAIP-2"]
```
Comment (bumblefudge) [inline mvx/README.md:9]:
```suggestion
```

File: mvx/caip2.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+---
+
+# CAIP-2
+
+_For context, see the [CAIP-2][] specification._
+
+## Abstract
+
+In CAIP-2 a general blockchain identification scheme is defined. This is the
+implementation of CAIP-2 for the MultiversX network.
```
Comment (bumblefudge) [inline mvx/caip2.md:19]:
```suggestion
profile of CAIP-2 specific to the MultiversX network.
```

File: mvx/caip2.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+## Rationale
+
+MultiversX consists of multiple networks: a production network (Mainnet), a development testing network (Devnet)
+and a testing network (Testnet).
+
+## Syntax
+
+An identifier for an MultiversX chain consists of the namespace prefix "mvx:"
+followed by the chain's Id.
```
Comment (bumblefudge) [inline mvx/caip2.md:33]:
```suggestion
followed by the appropriate chainId corresponding to that chain.
```

File: mvx/caip2.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+### Reference Definition
+
+The reference on the MultiversX chain is the equivalent of the MultiversX chainId:
```
Comment (bumblefudge) [inline mvx/caip2.md:37]:
```suggestion
The reference for each MultiversX chain is identical to its native representation in the MultiversX ecosystem:
```

File: mvx/caip2.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+`1` for `Mainnet`, `D` for `Devnet`, `T` for `Testnet`
+
```
Comment (bumblefudge) [inline mvx/caip2.md:39]: How are these values governed? Is there some kind of governance around future chainIds for future chains?  Will they always be single ASCII characters? Can someone spin up a private network and use some other character, and if so, would it be conformant to use, say, `X` if the dapp can learn (out of bound) how and where to connect to an X-network node?

File: mvx/caip10.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+---
+
+# CAIP-10
+
+_For context, see the [CAIP-10][] specification._
+
+## Rationale
+
+The MultiversX Address format is bech32, specified by the [BIP_0173][].
+The human-readable-part of the bech32 addresses is "erd" which means that the MultiversX address always starts with an `erd1` ("1" is a separator) e.g.: `erd1sea63y47u569ns3x5mqjf4vnygn9whkk7p6ry4rfpqyd6rd5addqyd9lf2`.
```
Comment (bumblefudge) [inline mvx/caip10.md:19]: Is 1 the separator only for mainnet, or are there addresses that begin `erdD` and `erdT` ?

File: mvx/caip10.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+The MultiversX network defines the address of an account as the bech32 representation of the public key of its corresponding pair of keys (the secret key remains known only to the user that owns the key pair). The public key is 32 bytes in length (64 hex characters), while the address (bech32) is 62 characters long.
```
Comment (bumblefudge) [inline mvx/caip10.md:21]: I assume this byte-->char conversion is base58btc encoding? Check out other CAIP-10 profiles for examples of how people define this dependency normatively.

File: mvx/caip10.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+On MultiversX, Smart Contracts and only Smart Contracts have their hex (raw) addresses prefixed with 8 zero bytes (meaning that their bech32 address will begin with `erd1qqqqqqqqqqqq` )
```
Comment (bumblefudge) [inline mvx/caip10.md:23]:
```suggestion
On MultiversX, deployed Smart Contracts (and only deployed Smart Contracts) have their hex (raw) addresses prefixed with 8 zero bytes (meaning that their bech32 address will begin with `erd1qqqqqqqqqqqq` )
```
How are deployed smart contracts queried or addressed, then?  Feel free to add a CAIP-19 profile as well if you have time :D

File: mvx/caip10.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+## Syntax
+
+```
+caip10-like address:    namespace + ":" chainId + ":" + address
```
Comment (bumblefudge) [inline mvx/caip10.md:28]: my only thought is that the `erq` and perhaps also the `1` are redundant, since mvx (and the fact that it's a CAIP-10) tells you it's a MultiversX account and `1` tells you (i presume?) that it's on mainnet.  Maybe you should conserve those 4 redundant characters in case some future form of mvx address is longer and you need the space?

File: mvx/caip10.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+namespace:              mvx
+chain Id:               1, D or T
+address:                bech32 formatted MultiversX address (erd1...)
+```
+
+## Test Cases
+
+```
+MultiversX Mainnet
```
Comment (bumblefudge) [inline mvx/caip10.md:37]:
```suggestion
wallet address on MultiversX Mainnet
```

File: mvx/caip10.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+mvx:1:erd1uapegx64zk6yxa9kxd2ujskkykdnvzlla47uawh7sh0rhwx6y60sv68me9
+mvx:1:erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3 ( Smart Contract )
```
Comment (bumblefudge) [inline mvx/caip10.md:39]:
```suggestion

#Deployed Smart Contract on MultiversX Mainnet
mvx:1:erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3 
```

**Comment by bumblefudge** [comment]
Hey there @radumojic -- thanks so much for opening this, interesting stuff!

Sorry for the delay/editorial outage, I should be more timely when you get back to me with answers on these questions.  I didn't really mark which were blocking needs and which were just suggestions or clarity edits, but I'd say this is most of the way there as the test cases seem good.  Think of these namespaces as a little more educational than pure specifications-- you're explaining the bare minimum someone needs to know to address or interact with chains and addresses *from outside* the ecosystem, so don't be afraid to ELI5.

Thanks!

File: mvx/caip10.md (hunk 6 above)
Comment (radumojic) [reply mvx/caip10.md:19]: Similar to Bitcoin SegWit bech32 addresses, `1` is the separator. For the moment, the only valid HRP is `erd` so testnet and devnet adresses will keep the same format of `erd1`

File: mvx/caip10.md (hunk 9 above)
Comment (radumojic) [reply mvx/caip10.md:28]: Although the first characters may seem redundant for now, it is a lot easier for developers to use the provided bech32 address instead of rebuilding it based on the namespace.

This also remains in sync with other namespaces as the provided address is easy to verify/get the balance of, etc without any other intervention.

File: mvx/caip10.md (hunk 7 above)
Comment (radumojic) [reply mvx/caip10.md:21]: No, the byte->char conversion is just the direct transformation from the public key bytes to the bech32 string representation: [https://en.bitcoin.it/wiki/Bech32](https://en.bitcoin.it/wiki/Bech32)

File: mvx/caip10.md (hunk 8 above)
Comment (radumojic) [reply mvx/caip10.md]: Sure, once a standard is agreed upon, further CAIPs will be added

File: mvx/caip2.md (hunk 5 above)
Comment (radumojic) [reply mvx/caip2.md:39]: No, the chainID value is not governed for the time being. Since there is only one mainnet chain (for now), the chainIDs for the test chains (testnet, devnet, etc) were decided socially so that the community could go with them.
More about the currently available constants can be found here: [https://docs.multiversx.com/developers/constants](https://docs.multiversx.com/developers/constants)
Once the current convention changes through a MIP(MultiversX Improvement Proposal) and a new standard is settled upon, this section will be updated.

**Comment by radumojic** [comment]
> Hey there @radumojic -- thanks so much for opening this, interesting stuff!
> 
> Sorry for the delay/editorial outage, I should be more timely when you get back to me with answers on these questions. I didn't really mark which were blocking needs and which were just suggestions or clarity edits, but I'd say this is most of the way there as the test cases seem good. Think of these namespaces as a little more educational than pure specifications-- you're explaining the bare minimum someone needs to know to address or interact with chains and addresses _from outside_ the ecosystem, so don't be afraid to ELI5.
> 
> Thanks!

Hello, 
Thank you for your time and your feedback! I've applied the suggestions from the code review and hopefully answered your questions.
Once more conventions are settled upon through MIPs(MultiversX Improvement Proposals) further CAIPs will be added ( CAIP-19 and CAIP-122 come first to mind )

Please let me know if there's anything else I can help with

File: mvx/caip10.md (hunk 7 above)
Comment (bumblefudge) [reply mvx/caip10.md:21]: whoops my mistake had bech32 mixed up with something else in my mind 😅

**Comment by bumblefudge** [comment]
LGTM!  Next CASA editor to double-check my work will merge it in.  Thanks again for taking the time to spell things out that should have been obvious, hehe.
